### DISCUSSION TITLE
USB Serial on Synology NAS

### DISCUSSION THREAD

**Discussion by GitMarkLab** [OctoPrint/docker#193] [Q&A]
Hi,

i would like use octoprint-docker on a Synology DS720+ with an Anycubic.
Now i have problems to establish the USB Serial communication.

I logged in into the NAS via SSH and add the following comments

sudo insmod /lib/modules/usbserial.ko
sudo insmod /lib/modules/ftdi_sio.ko
sudo chmod 777 /lib/modules/usbserial.ko
sudo chmod 777 /lib/modules/ftdi_sio.ko

(https://community.openhab.org/t/get-serial-devices-working-on-synology-ds216-ii-diskstation/36072/4)

than i checked /dev/tty* but no ttyUSB0 appered. Is it correct to establish the connection on the NAS first and than integrate it 
into octoprint-docker?

Any ideas how to set it up?

BR

**Comment by GitMarkLab** [comment]
ok the printer needs an additional custom driver

UsbSerialDrivers DSM 6.2 v6-8

http://www.jadahl.com/drivers_6.2/?arch=geminilake



it now works for me with a docker-compose file. Just add
    ports:
      - 32772:80
    devices:
     - /dev/ttyUSB0:/dev/ttyUSB0

where 32772 is custom. I will use a network cam, maybe ESP32Cam....

**Reply by ImaBadCoder** [reply]
Does this work for DSM 7.0 as well?

**Reply by GitMarkLab** [reply]
I did not try it. I switched back to separate PIs because of instability (connection lost)

**Reply by ImaBadCoder** [reply]
I spent several hours on it, got the ttyUSB0 to show up, but could not get the Container to see it.. So I transitioned to using Octoprint as an Addon in Home Assistant.
